R/client-cloud.R

```diff
@@ -66,20 +66,46 @@ cloudClient <- function(service, authInfo) {
       listRequest(service, authInfo, path, query, "applications")
     },
 
-    getApplication = function(applicationId) {
-      path <- paste("/applications/", applicationId, sep = "")
-      application <- GET(service, authInfo, path)
+    getApplication = function(outputOrApplicationId, dcfVersion) {
+      if (is.na(dcfVersion)) {
```

Comment:
I'm passively curious why versioning ended up being favored over the kinda duck-typing strategy where `outputOrApplicationId` was a lucid resource identifier.

Is this PR intended to address https://github.com/rstudio/rsconnect/issues/819?

Reply:
I tested my understanding of the acceptance criteria for #819 using this branch, but get various errors or misbehaviors (new output in Cloud) depending on whether I use `deployDoc` or `deployApp` and whether an `rsconnect` directory is present.

I'm not sure fixing #819 is in scope for this PR, but I'm unsure this PR really resolves it. Does it work for you? Should we reopen that one?

Reply:
Ah, I misunderstood the problem... I had interpreted it as being related to the in-progress work for this issue, but it's actually a regression on main.